src/psyclone/psyGen.py

```diff
@@ -3965,25 +3955,20 @@ def names(self):
     def args(self):
         return self._args
 
-    def iteration_space_arg(self, mapping={}):
+    def iteration_space_arg(self):
         '''
         Returns an argument that can be iterated over, i.e. modified
         (has WRITE, READWRITE or INC access).
 
-        :param mapping: dictionary of access types associated with arguments
-                        with their metadata strings as keys
-        :type mapping: dict
         :returns: a Fortran argument name
         :rtype: string
         :raises GenerationError: if none such argument is found.
 
         '''
-        assert mapping != {}, "psyGen:Arguments:iteration_space_arg: Error "
-        "a mapping needs to be provided"
         for arg in self._args:
-            if arg.access.lower() == mapping["write"] or \
-               arg.access.lower() == mapping["readwrite"] or \
-               arg.access.lower() == mapping["inc"]:
+            if arg.access == AccessType.WRITE or \
```

Comment:
Could just do something like:

    if arg.access not in AccessType.write_accesses():

perhaps?

Reply:
OK, now I had that problem in several places: why is 'sum' missing in that list? AccessType has a 'all_write_access()' function, but it includes the reductions (i.e. sum). If we can add SUM here, then it's easy. If not, we need to find better names, which might get quite confusing ('all_write_access', 'all_write_non_reduction_accesses', ...).

Reply:
Off the top of my head I don't know the answer. Perhaps @rupertford can remember? I suspect we could put SUM in the list and see what happens...

Reply:
I started to have a look at this. From the docs it would indeed appear that a reduction should not be in that list (it is used to find a field to iterate over - in case of a SUM I'd guess the argument would be a scalar). So for now I am using existing AccessType functions:
```
            if arg.access in AccessType.all_write_accesses() and \
                    arg.access not in AccessType.get_valid_reduction_modes():
                return arg
```
I also tried to add a test case for this, but it turns out that dynamo3 does not use this function (it overwrites it with its own function and does not call the function in psyGen), and of course gocean does not support SUM in the first place. I'll just left it unchanged (except for the lines above, which I think will indeed increase portability later).